CONTRIBUTING.md

```diff
@@ -124,3 +124,73 @@ what a commit is about.
 
 When you create a pull request, please
 [make your PR editable](https://github.com/blog/2247-improving-collaboration-with-forks).
+
+## Rebasing
+
+If other PRs have been merged during the time between your initial PR creation
+and final approval, it may be required that you rebase your changes against the
+latest `main` branch.
+
+There are potential pitfalls here if you follow the suggestions from `git`,
+which could leave your branch in an unrecoverable mess,
+and you having to start over with a new branch and new PR.
+
+The procedure below is tried and tested, and will help you avoid frustration.
+
+To rebase a feature branch to the latest `main`:
+
+1. Make sure that your local copy of the repository has the most up-to-date
+   revisions of `main` (this is important, otherwise you may end up rebasing to
+   an older base point):
+   ```bash
+   git switch main
+   git pull
+   ```
```

Comment:
Switching branches is not necessary and incurs some overhead (updating the worktree twice). We could just instruct to use `git fetch` and then later merge with with the fetched ref (e.g. `origin/main`).

Reply:
As a `git` novice, I found a workflow that worked, and documented that.

Having now read [this](https://stackoverflow.com/questions/292357/what-is-the-difference-between-git-pull-and-git-fetch), I can see that `fetch` might be more useful if multiple people are making changes to a PR - but then you have to run `merge` as a separate step.  And most PRs are an individual contributor.

It's not clear to me if either `pull` or `fetch` will update all branches or only the currently active one.  Hence I went for a more robust approach that was guaranteed to work.

Reply:
> [Git pull is a Git command that performs both git fetch and git merge simultaneously.](https://about.gitlab.com/blog/2024/09/24/git-pull-vs-git-fetch-whats-the-difference/)

So `git fetch` updates (local) remote tracking branches (i.e., "shadow copies" of the remote branches), while `git pull` updates a local branch by doing a `fetch` and then a  `merge`.